src/library_pthread.js

```diff
@@ -159,11 +162,11 @@ var LibraryPThread = {
 #if PTHREADS_DEBUG
       err('terminateAllThreads');
 #endif
-      for (var t in PThread.pthreads) {
-        var worker = PThread.pthreads[t];
-        if (worker) {
-          PThread.returnWorkerToPool(worker);
-        }
+      for (var worker of Object.values(PThread.pthreads)) {
```

Comment:
Could change PThreads.pthreads to a `Map` then it would behave more like it's used in this file. That would require some more changes though.

Reply:
Will try that now... might make it a followup.. lets see.

Reply:
I tried using `Map` and it was a slight codesize regression, so I'd rather land this change as is (at least for now)